plasma/morevp_2.0.md

```diff
@@ -0,0 +1,109 @@
+2. Alice can piggyback on her output `o4` out of that. Note this is a "regular" output and she can't sign spending of `o4` or she will be challenged MoreVP IFE-style.
+3. Exchange is not finished with the `super_tx` at hand, so it chains the next tx to that: `E.chain(tx1, 1, tx2)`, indicating that 1st output `o5` of `tx1` has been consumed by a follow-up `tx2`.
+(Think of this as a "mixture" of a `startIFE` and `piggyback_output` message.)
+Observe that `tx2` consumes `o3` and `o5` as inputs: if this were a `startIFE` we'd require `E` to deliver inclusion proof of txs creating those `o3`, `o5`, but it is **only necessary** for `o3` - `o5` is chained instead of proven
+4. Here Bob and Charlie do what Alice did in (2/)
+
+If there were more txs to be chained `tx3`, `tx4`, ... the process would iterate naturally.
+
+`super_tx` here is `[tx1, tx2]`, it is a state transition from `o1`, `o2`, `o3` to `o4`, `o6`, `o7`.
+
+### Canonicity game and the `super_tx`
+
+#### General flow
+
+The "happened or not" of the `super_tx` is dependent on the canonicity of the individual chained txs.
+If both `tx1` and `tx2` are canonical, the `super_tx` happened in its entirety (`o4`, `o6`, `o7` should exit)
+If `tx2` turns out to be non-canonical, the `super_tx` happened only as far as `tx1` goes (`o3`, `o4`, `o5` should exit)
+If `tx1` turns out to be non-canonical, the `super_tx` didn't happen at all (`o1-3` should exit).
+
+#### Canonicity index
+
+One might think that this would require the contract to iterate over the chain of exits to change the status of the `super_tx`, as individual canonicities of the chained txs change.
+This is not the case - the exiting data can be represented in a way that allows this to be known without iterating.
+This is done by introducing the concept of `canonicity_index`.
+The "happened or not" of the `super_tx` is not represented by a boolean (canonical/non-canonical), but by an integer `n` meaning, `n` chained txs are "canonical".
+If there's an `n+1`th transaction then it is the first non-canonical in the chain.
+
+Using `canonicity_index`, one can organize the exiting funds of the `super_tx` so that on `processExits`, the cIFE "knows" which inputs and outputs to allow out.
+
+#### Requirements on competitors
+
+To ensure that the canonicity game makes sense, we are after the following property.
+There must be only 2 ways a chain of txs may have been attempted to be executed:
+ - by publishing by `E` in a cIFE
+ - by including in child chain blocks
+
+If `E` has double-signed some settlements, the version included in valid child chain blocks prevails and is published.
+If `E` didn't double-sign, the single chain of txs is the one in the cIFEs, will be canonical.
+
+In both cases the users have access to the funds.
+
+##### Proof for competitors required
+
+An important modification of the canonicity game we introduce for the chained txs is that submitting a competitor **requires an inclusion proof**.
```

Comment:
I am lost here actually, why inclusion proof is required?

Reply:
It simplifies the lifecycle of such cIFE, a simplification made possible thanks to a single actor driving the process here (`E`). Once un-canonicalized, those txs from cIFE won't "matter" anymore.

For example, you don't have to bother with the "prove canonical" part. 

But if this requirement (or the assumptions is uses) bother us too much, there likely is a way to remove it